src/buildzqfield.js

```diff
@@ -46,9 +48,27 @@ class ZqBuilder {
 
 }
 
+function is_adx_supported() {
+
+    try {
+        data = fs.readFileSync('/proc/cpuinfo', 'utf8');
+        const adxSupported = data.includes('adx');
+        if (adxSupported) {
+            return true;
+        } else {
+            return false;
+        }
+    } catch (err) {
+        return false;
+    }
+}
+
 async function buildField(q, name, no_adx, hpp_out, element_hpp_out) {
-    const builder = new ZqBuilder(q, name, no_adx, hpp_out, element_hpp_out);
 
+    if (!no_adx) {
+        no_adx = !is_adx_supported();
+    }
```

Comment:
then flag would become not optional, this lines checks if no flag was send if it wasnt detect and set it if false/true

actually I have lines  in the pull 

if (no_adx == null) {
        no_adx = !is_adx_supported();
    }

Reply:
testing this allows

1. no_adx option , detection will be performed to avoid illegal ops
2. --no_adx true or no_adx false will not perform detection and honor the override 

(no_adx == null) ? no_adx = !is_adx_supported() : no_adx = Boolean(no_adx.toLowerCase() === "true");